src/ui/components/Transcript/TranscriptItem/FloatingTranscriptItem.tsx

```diff
@@ -1,43 +1,44 @@
-import React, { useState, useEffect } from "react";
+import React from "react";
 import { connect, ConnectedProps } from "react-redux";
 
 import CommentThread from "ui/components/Comments/TranscriptComments/CommentThread";
 import TranscriptItem from "./TranscriptItem";
-import { PendingComment, PauseItem, SourceLocation } from "ui/state/comments";
-import { ThreadFront } from "protocol/thread";
+import { FloatingItem } from "ui/state/comments";
 import { UIState } from "ui/state";
+import { selectors } from "ui/reducers";
 
 const { getFilenameFromURL } = require("devtools/client/debugger/src/utils/sources-tree/getURL");
 const { getTextAtLocation } = require("devtools/client/debugger/src/reducers/sources");
 const { getFormattedTime } = require("ui/utils/timeline");
+const { findClosestFunction } = require("devtools/client/debugger/src/utils/ast");
+const { getSymbols } = require("devtools/client/debugger/src/reducers/ast");
 
-type PauseTranscriptItemProps = PropsFromRedux & {
-  item: PauseItem;
-  pendingComment: PendingComment | null;
+type PropsFromParent = {
+  item: FloatingItem;
 };
 
+type PauseTranscriptItemProps = PropsFromRedux & PropsFromParent;
+
 // Transcript item component for displaying a temporary pause point that is not already
 // accounted for by one of the displayed entries.
 
-function PauseTranscriptItem({ item, pendingComment, state }: PauseTranscriptItemProps) {
-  const [sourceLocation, setSourceLocation] = useState<SourceLocation | null>(null);
+function PauseTranscriptItem({
+  item,
+  pendingComment,
+  snippet,
+  closestFunction,
+}: PauseTranscriptItemProps) {
   let icon = "location-marker";
   let label = "Point In Time";
   let secondaryLabel = getFormattedTime(item.time);
 
-  useEffect(() => {
-    const getSourceLocation = async () => {
-      const location = (await ThreadFront.getCurrentPauseSourceLocation()) || null;
-      setSourceLocation(location);
-    };
-    getSourceLocation();
-  }, [item]);
-
-  if (sourceLocation) {
-    const filename = getFilenameFromURL(sourceLocation.sourceUrl);
+  if (item.location) {
+    // const { sourceUrl, line } = item.location;
+    // const filename = getFilenameFromURL(sourceUrl);
+    // label = `${filename}:${line}`;
     icon = "document-text";
-    label = `${filename}:${sourceLocation.line}`;
-    secondaryLabel = getTextAtLocation(state, sourceLocation.sourceId, sourceLocation);
+    label = closestFunction?.name || "(anonymous)";
```

Comment:
`findClosestFunction` function either returns 1) an object that represents the function, or 2) null.

This just makes sure that in null cases, we just refer to it as `(anonymous)`.

Reply:
How about we use the function name if we have it otherwise fall back to the source filename